src/wp-includes/comment-template.php

```diff
@@ -2713,15 +2713,44 @@ function comment_form( $args = array(), $post = null ) {
 
 		else :
 
+			/**
+			 * Filters additional comment form attributes.
+			 *
+			 * The following attributes are allowed:
+			 * `accept-charset`, `autocapitalize`, `autocomplete`, `enctype`, `novalidate`
+			 * Pass attribute names as the array key.
+			 * Pass attribute values as the array value, or `true` in the case of attributes that require no value, for example `novalidate`.
+			 *
+			 * @since CP-2.1.0
+			 *
+			 * @param array   An associative array containing the additional attributes for the form.
+			 */
+			$comment_form_attrs = (array) apply_filters( 'comment_form_attrs', array() );
+
+			$comment_form_attributes = '';
+			if ( ! empty( $comment_form_attrs ) ) {
+				$allowed_attributes = array( 'accept-charset', 'autocapitalize', 'autocomplete', 'enctype', 'novalidate' );
+				foreach ( $comment_form_attrs as $comment_form_attr_name => $comment_form_attr_value ) {
+					if ( in_array( $comment_form_attr_name, $allowed_attributes ) ) {
+						if ( 'true' === $comment_form_attr_value ) {
```

Comment:
Why checking for the `true` string and not for a boolean value?
None of those attributes accept `true` as a valid value, but what if a new attribute will use `true` as a valid value?

Reply:
I did consider having a  boolean check but casting to boolean on any string will evaluate to `true`. As all other keys and values passed were strings I'd thought to ask filter users to pass everything as a string, includes `'true'` for attributes that don't need a key for this filter.

Reply:
I think having an array with values of different type is pretty normal, but if you prefer to do this way I think that docblock should say "true" instead of 
```
`true`
```

 so that is more clear that is a string.
Because if you write `'novalidate'=>true` the result is `novalidate="1"`.